azure-iot-device/azure/iot/device/iothub_session.py

```diff
@@ -379,6 +379,18 @@ async def wrapping_coroutine():
 
         return wrapping_coroutine()
 
+    @property
```

Comment:
Lack of a connected attribute forces a particular programming style. Saying "the customer _shouldn't_ care" is OK from a purity standpoint, but customers _will_ care. 

Without this, I would call the session object "just like a normal connection object (or file handle or open socket, or file descriptor) except you can't know if it's connected unless you keep track of it yourself.

Consider a case where the main function is waiting for method requests and a task is used to send telemetry every 5 seconds.

```
async def main():
    with IoTHub.create_from_connection_string() as session:
        # create a task to send telemetry every 4 seconds:
        sender_task = asyncio.create_task(sender_function(session))

        # handle direct method requests
        async with session.direct_method_requests() as requests:
            for request in requests:
                handle_request(requests)

async def sender_function(session):
    while True:
        telemetry = await get_data_from_sensor()
        await session.send_message(telemetry)
        await asyncio.sleep(5)
```

This won't work because `sender_function` doesn't have any way to know if the session is connected or not. I guess it could wrap `send_message` in a try/except block. and that would fix this scenario, but it requires the developer to attempt a send_message call before it can know the connection state.

But what if they want to know the connection state without sending a message? Our library knows if we're connected or not, but why would we refuse to tell them?

Reply:
Yeah, okay, fair enough. I could see the case with unawaited tasks.

Reply:
thanks. I didn't see `self._mqtt_client.connected` the first time I went through.